### DISCUSSION TITLE
Grain average Image Quality MTEX

### DISCUSSION THREAD

**Discussion by Mabroukimedcea** [mtex-toolbox/mtex#1812] [Ask Anything]
Hello everyone, 

I am trying to plot the distribution of the grain Image Qualtity with MTEX ( like the one below) , 
Does anyone know how we can do it please? if there is a function that is ablel to do it directly 

Thanks in advanced 
![image](https://user-images.githubusercontent.com/126521440/236400363-13794194-1747-415f-8e8e-48f943a0c3bf.png)

**Comment by Mabroukimedcea** [comment]
For information 
I have calculated the grains from my EBSD_data 
as bleow 
[grains,EBSD_data.grainId,MB_EBSD.mis2mean]  = calcGrains(EBSD_EBSD,'angle',threshold);

and then I didn't get how to get the Image qualtiy per grain 

with the EBSD_data I get Image_quality per pixel

**Comment by kilir** [comment]
Hi,
´grainMean´ should do that.  
```
[grains,ebsd.grainId]=calcGrains(ebsd)
meanProp = grainMean(ebsd, ebsd.iq); % or whatever the image quality is called
```
Cheers,
Rüdiger

**Reply by Mabroukimedcea** [reply]
thank you so much Rüdiger

**Reply by Mabroukimedcea** [reply]
> Hi, ´grainMean´ should do that.
> 
> ```
> [grains,ebsd.grainId]=calcGrains(ebsd)
> meanProp = grainMean(ebsd, ebsd.iq); % or whatever the image quality is called
> ```
> 
> Cheers, Rüdiger

One more question please  Rüdiger
Can we choose a group of grains by their Image quality ? 
thanks in advanced 
Mohamed

**Reply by Mabroukimedcea** [reply]
> > Hi, ´grainMean´ should do that.
> > ```
> > [grains,ebsd.grainId]=calcGrains(ebsd)
> > meanProp = grainMean(ebsd, ebsd.iq); % or whatever the image quality is called
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Cheers, Rüdiger
> 
> One more question please Rüdiger Can we choose a group of grains by their Image quality ? thanks in advanced Mohamed

I just get it 
 i have to define my condition 
and then 
new_grains = grains (my_conditions); 


thanks anyway and have a great weekend
